tests/test_htcondor_service.py

```diff
@@ -532,3 +536,487 @@ def testRelPathId(self, _wms_id_type_mock):
+            "8624> fired when job (8624.0) was removed",
+            "job_held_time": "2025-02-07T12:33:34",
+            "job_released_time": "2025-02-07T12:33:47",
+        }
+        results = WmsSpecificInfo()
+        _add_service_job_specific_info(job_ad, results)
+        self.assertEqual(
+            results.context, {"job_name": "provisioningJob", "status": "SUCCEEDED", "status_details": ""}
+        )
+
+    def testHeldReleasedDeleted(self):
+        # Since held info will be in job_ad,
+        # make sure knows status after released.
+        job_ad = {
+            "ClusterId": 9086,
+            "DAGNodeName": "provisioningJob",
+            "HoldReason": "via condor_hold (by user mgower)",
+            "HoldReasonCode": 1,
+            "HoldReasonSubCode": 0,
+            "JobStatus": JobStatus.REMOVED,
+            "ProcId": 0,
+            "Reason": "via condor_rm (by user mgower)",
+            "job_evicted_time": "2025-02-11T11:35:04",
+            "job_held_time": "2025-02-11T11:35:04",
+        }
+        results = WmsSpecificInfo()
+        _add_service_job_specific_info(job_ad, results)
+        self.assertEqual(
+            results.context, {"job_name": "provisioningJob", "status": "DELETED", "status_details": ""}
+        )
+
+    def testHeldReleasedHeld(self):
+        # Since release info will be in job_ad,
+        # make sure knows held after release.
+        job_ad = {
+            "ClusterId": 8659,
+            "DAGNodeName": "provisioningJob",
+            "HoldReason": "via condor_hold (by user mgower)",
+            "HoldReasonCode": 1,
+            "HoldReasonSubCode": 0,
+            "JobStatus": JobStatus.REMOVED,
+            "ProcId": 0,
+            "Reason": "Removed by DAGMan (by user mgower)",
+            "TerminatedNormally": False,
+            "job_held_time": "2025-02-07T12:36:15",
+            "job_released_time": "2025-02-07T12:36:07",
+        }
+        results = WmsSpecificInfo()
+        _add_service_job_specific_info(job_ad, results)
+        self.assertEqual(
+            results.context,
+            {
+                "job_name": "provisioningJob",
+                "status": "HELD",
+                "status_details": "(via condor_hold (by user mgower))",
+            },
+        )
+
+
+class GetRunSummaryTestCase(unittest.TestCase):
+    """Test _get_run_summary function."""
+
+    def testJobSummaryInJobAd(self):
+        summary = "pipetaskInit:1;label1:2;label2:2;finalJob:1"
+        job_ad = {"ClusterId": 8659, "DAGNodeName": "testJob", "bps_job_summary": summary}
+        results = _get_run_summary(job_ad)
+        self.assertEqual(results, summary)
+
+    def testRunSummaryInJobAd(self):
+        summary = "pipetaskInit:1;label1:2;label2:2;finalJob:1"
+        job_ad = {"ClusterId": 8659, "DAGNodeName": "testJob", "bps_run_summary": summary}
+        results = _get_run_summary(job_ad)
+        self.assertEqual(results, summary)
```

Comment:
Looks like a duplicate of the first test. Please, remove.

Reply:
It isn't a duplicate.  They are checking for 2 different keys in the job ad (job vs run).

Reply:
Right, I missed that.